src/collection/evaluate_collection.py

```diff
@@ -0,0 +1,350 @@
+from src.collection.query_collection import get_top_k_results, filter_search
+from sentence_transformers import SentenceTransformer
+from typing import List
+from qdrant_client import QdrantClient
+from src.utils.bigquery import query_bigquery
+import regex as re
+
+
+def load_qdrant_client(qdrant_host: str, port: int) -> QdrantClient:
+    client = QdrantClient(qdrant_host, port=port)
+    return client
+
+
+def load_model(model_name: str) -> SentenceTransformer:
+    """
+    Load the SentenceTransformer model.
+
+    Args:
+        model_name (str): The name of the model.
+
+    Returns:
+        SentenceTransformer: The loaded model.
+    """
+    model = SentenceTransformer(model_name)
+    return model
+
+
+def calculate_precision(retrieved_records: list, relevant_records: int) -> float:
+    """
+    Calculate precision
+
+    Args:
+        retrieved_records (int): Number of retrieved records
+        relevant_records (int): Number of relevant records
+
+    Returns:
+        float: Precision
+    """
+    true_positives = len(set(retrieved_records).intersection(relevant_records))
+    return true_positives / len(retrieved_records) if retrieved_records else 0
+
+
+def calculate_recall(retrieved_records: dict, relevant_records: int) -> float:
+    """
+    Calculate recall
+
+    Args:
+        retrieved_records (int): Number of retrieved records
+        relevant_records (int): Number of relevant records
+
+    Returns:
+        float: Recall
+    """
+    true_positives = len(set(retrieved_records).intersection(relevant_records))
+    return true_positives / len(relevant_records) if relevant_records else 0
+
+
+def calculate_f1_score(precision: dict, recall: int) -> float:
+    """
+    Calculate F1 score
+
+    Args:
+        precision (float): Precision
+        recall (float): Recall
+
+    Returns:
+        float: F1 score
+    """
+    return (
+        2 * (precision * recall) / (precision + recall)
+        if (precision + recall) > 0
+        else 0
+    )
+
+
+def calculate_f2_score(precision: dict, recall: int) -> float:
+    """
+    Calculate F2 score
+
+    Args:
+        precision (float): Precision
+        recall (float): Recall
+
+    Returns:
+        float: F2 score
+    """
+    return (
+        5 * (precision * recall) / (4 * precision + recall)
+        if (4 * precision + recall) > 0
+        else 0
+    )
+
+
+def get_data_for_evaluation(
+    evaluation_table: str,
+    project_id: str,
+) -> List[dict]:
+    """
+    Query BQ for labelled feedback data. for use in evaluation.
+
+    Returns:
+        list(dict): Feedback data IDs and labels
+    """
+    query = """
+    SELECT
```

Comment:
Might be cleaner to have this in src/sql_queries.py, to keep all queries in the same place (e.g. if table schemas / metadata change)

Reply:
Created issue